src/Providers/Http/Util/ResponseUtil.php

```diff
@@ -33,35 +33,6 @@ public static function throwIfNotSuccessful(Response $response): void
             return;
         }
 
-        $errorMessage = sprintf(
-            'Bad status code: %d.',
-            $response->getStatusCode()
-        );
-
-        // Handle common error formats in API responses.
-        $data = $response->getData();
-        if (
-            is_array($data) &&
-            isset($data['error']) &&
-            is_array($data['error']) &&
-            isset($data['error']['message']) &&
-            is_string($data['error']['message'])
-        ) {
-            $errorMessage .= ' ' . $data['error']['message'];
-        } elseif (
-            is_array($data) &&
-            isset($data['error']) &&
-            is_string($data['error'])
-        ) {
-            $errorMessage .= ' ' . $data['error'];
-        } elseif (
-            is_array($data) &&
-            isset($data['message']) &&
-            is_string($data['message'])
-        ) {
-            $errorMessage .= ' ' . $data['message'];
-        }
-
-        throw new ResponseException($errorMessage, $response->getStatusCode());
+        throw ResponseException::fromBadResponse($response);
```

Comment:
I'm thinking that we may be want to move the `fromBadResponse` method to `RequestException`? Technically, the lines are blurry.

The only thing I personally consider a response exception is when something in the response itself is wrong (e.g. missing an expected field in a successful response), which should almost never happen. Everything else IMO is more suitable in `RequestException` - because if the response has an error status code, we can consider the request failed, not the response.

Would love to get your perspective on this too @JasonTheAdams!

Reply:
So in frameworks like Symfony they distinguish this a bit differently, they use:
* `HttpExceptionInterface`, which indicates a 4xx or 5xx response error
* `ClientExceptionInterface`, which extends `HttpExceptionInterface` and is specific to 4xx response errors
* `ServerExceptionInterface`, which extends `HttpExceptionInterface` and is specific to 5xx response errors
* `TransportExceptionInterface`, which means there was a network failure

So they don't actually distinguish it in terms of "response" and "request", I think nbecause of the ambiguity being described here. So if we like that convention better we could go that route.

Reply:
I see your point. I'm good with `RequestException`, or if we want something a touch more granular: `UnsupportedRequestException`.